src/plugins/navigation_embeddable/public/components/navigation_embeddable_link_editor.tsx

```diff
@@ -0,0 +1,175 @@
+          color="text"
+          iconType={'arrowLeft'}
+          onClick={() => onClose(false)}
+        >
+          <EuiTitle
+            size="m"
+            aria-label={NavEmbeddableStrings.editor.linkEditor.getGoBackAriaLabel()}
+          >
+            <h2>{NavEmbeddableStrings.editor.getAddButtonLabel()}</h2>
+          </EuiTitle>
+        </EuiButtonEmpty>
+      </EuiFlyoutHeader>
+      <EuiFlyoutBody>
+        <EuiForm component="form" fullWidth>
+          <EuiFormRow label={NavEmbeddableStrings.editor.linkEditor.getLinkTypePickerLabel()}>
+            <EuiRadioGroup
+              options={linkTypes}
+              idSelected={selectedLinkType}
+              onChange={(id) => {
+                setLinkDestination(undefined);
+                setLinkLabelPlaceholder(undefined);
+                setSelectedLinkType(id as NavigationLinkType);
+              }}
+            />
+          </EuiFormRow>
+
+          <EuiFormRow label={NavEmbeddableStrings.editor.linkEditor.getLinkDestinationLabel()}>
+            {selectedLinkType === DASHBOARD_LINK_TYPE ? (
+              <DashboardLinkDestinationPicker
+                parentDashboard={parentDashboard}
+                setDestination={setLinkDestination}
+                currentDestination={linkDestination}
+                setPlaceholder={setLinkLabelPlaceholder}
+              />
+            ) : (
+              <ExternalLinkDestinationPicker
+                setDestination={setLinkDestination}
+                currentDestination={linkDestination}
+                setPlaceholder={setLinkLabelPlaceholder}
+              />
+            )}
+          </EuiFormRow>
+
+          <EuiFormRow label={NavEmbeddableStrings.editor.linkEditor.getLinkTextLabel()}>
+            <EuiFieldText
+              placeholder={
+                linkLabelPlaceholder ||
+                NavEmbeddableStrings.editor.linkEditor.getLinkTextPlaceholder()
+              }
+              value={linkLabel}
+              onChange={(e) => {
+                setLinkLabel(e.target.value);
+              }}
+            />
+          </EuiFormRow>
+        </EuiForm>
+
+        {/* TODO: As part of https://github.com/elastic/kibana/issues/154381, we should pull in the custom settings for each link type.
+            Refer to `x-pack/examples/ui_actions_enhanced_examples/public/drilldowns/dashboard_to_discover_drilldown/collect_config_container.tsx`
+            for the dashboard drilldown settings, for example. 
+
+            Open question: It probably makes sense to re-use these components so any changes made to the drilldown architecture
+            trickle down to the navigation embeddable - this would require some refactoring, though. Is this a goal for MVP?
+         */}
+      </EuiFlyoutBody>
+      <EuiFlyoutFooter>
+        <EuiFlexGroup responsive={false} justifyContent="spaceBetween">
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty onClick={() => onClose(true)} iconType="cross">
```

Comment:
I think we shouldn't close the flyout when canceling the Add link editor. Otherwise we risk the user losing any other links they've already added. 

```suggestion
            <EuiButtonEmpty onClick={() => onClose(false)} iconType="cross">
```

Reply:
Yeah, that's a very fair point! I kind of went back and forth on this behaviour in the first place :) Fixed!